Error when executing "npm run build-android-bundle" with 3.1.1 on a project that is located in a directory path with "spaces"

When I run: "npm run build-android-bundle", I see following error:

Project successfully prepared (android)
Executing after-prepare hook from /Users/mivanova/Documents/Telerik/UI for NativeScript 3.0.1.304/Examples/SDKAngular/hooks/after-prepare/nativescript-unit-test-runner.js
/bin/sh: /Users/mivanova/Documents/Telerik/UI: No such file or directory
child process exited with code 127
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! nativescript-telerik-ui-sdk-angular-examples@3.0.0 ns-bundle: `ns-bundle`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the nativescript-telerik-ui-sdk-angular-examples@3.0.0 ns-bundle script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mivanova/.npm/_logs/2017-07-03T12_08_18_366Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nativescript-telerik-ui-sdk-angular-examples@3.0.0 build-android-bundle: `npm run ns-bundle --android --build-app`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nativescript-telerik-ui-sdk-angular-examples@3.0.0 build-android-bundle script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Here is my **package.json** file: 
"dependencies": {
    "@angular/common": "~4.0.0",
    "@angular/compiler": "~4.0.0",
    "@angular/core": "~4.0.0",
    "@angular/forms": "~4.0.0",
    "@angular/http": "~4.0.0",
    "@angular/platform-browser": "~4.0.0",
    "@angular/platform-browser-dynamic": "~4.0.0",
    "@angular/router": "~4.0.0",
    "nativescript-angular": "~3.1.0",
    "nativescript-fresco": "^3.0.2",
    "nativescript-intl": "^3.0.0",
    "nativescript-telerik-ui-pro": "../../packages/nativescript-ui-pro.tgz",
    "nativescript-unit-test-runner": "^0.3.2",
    "rxjs": "~5.3.0",
    "tns-core-modules": "3.1.0",
    "zone.js": "~0.8.2"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~4.0.0",
    "@ngtools/webpack": "~1.4.0",
    "babel-traverse": "6.24.1",
    "babel-types": "6.24.1",
    "babylon": "6.17.0",
    "chai": "^3.5.0",
    "codelyzer": "^3.0.0",
    "copy-webpack-plugin": "~4.0.1",
    "extract-text-webpack-plugin": "~2.1.0",
    "filewalker": "0.1.3",
    "karma": "^1.6.0",
    "karma-chai": "^0.1.0",
    "karma-mocha": "^1.3.0",
    "karma-nativescript-launcher": "^0.4.0",
    "lazy": "1.0.11",
    "mocha": "^3.3.0",
    "nativescript-css-loader": "~0.26.0",
    "nativescript-dev-typescript": "^0.5.0",
    "nativescript-dev-webpack": "0.7.1",
    "raw-loader": "~0.5.1",
    "resolve-url-loader": "~2.0.2",
    "tns-platform-declarations": "~3.1.0",
    "tslint": "^5.1.0",
    "typescript": "~2.3.2",
    "webpack": "~3.0.0",
    "webpack-bundle-analyzer": "^2.8.2",
    "webpack-sources": "~1.0.1"
  },
  "scripts": {
    "ns-bundle": "ns-bundle",
    "tslint": "tslint --project tsconfig.json --config tslint.json",
    "publish-ios-bundle": "npm run ns-bundle --ios --publish-app",
    "generate-android-snapshot": "generate-android-snapshot --targetArchs arm,arm64,ia32 --install",
    "start-android-bundle": "npm run ns-bundle --android --run-app",
    "start-ios-bundle": "npm run ns-bundle --ios --run-app",
    "build-android-bundle": "npm run ns-bundle --android --build-app",
    "build-ios-bundle": "npm run ns-bundle --ios --build-app"
  }

--- a/bin/ns-bundle
+++ b/bin/ns-bundle
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 const { spawn } = require("child_process");
-const { resolve: pathResolve } = require("path");
+const { resolve: pathResolve, join } = require("path");
 const { existsSync } = require("fs");
 const { getPackageJson } = require("../projectHelpers");
 const { isVersionGte } = require("../utils");
@@ -13,14 +13,16 @@ if (!process.env.npm_config_argv) {
     throwError({message: "No flags provided."});
 }
 
-const escape = arg => `"${arg}"`;
+const escapeWithQuotes = arg => `"${arg}"`;
 const isTnsCommand = flag => flag.endsWith("-app");
 const isEnvCommand = flag => flag.indexOf("env.") > -1;
 const shouldUglify = () => process.env.npm_config_uglify;
-const shouldSnapshot = (platform) => platform == "android" && require("os").type() != "Windows_NT" && process.env.npm_config_snapshot;
+const shouldSnapshot = platform => platform == "android" &&
+    require("os").type() != "Windows_NT" &&
+    process.env.npm_config_snapshot;
 
 const npmArgs = JSON.parse(process.env.npm_config_argv).original;
-const tnsArgs = getTnsArgs(npmArgs).map(escape);
+const tnsArgs = getTnsArgs(npmArgs).map(escapeWithQuotes);
 const flags = npmArgs.filter(a => a.startsWith("--")).map(a => a.substring(2));
 const options = getOptions(flags);
 
@@ -51,7 +53,7 @@ function execute(options) {
             ...commands,
             () => cleanApp(platform),
             () => cleanSnapshotArtefacts(),
-            () => cleanBuildArtifacts(platform),
+            () => cleanBuildArtefacts(platform),
             () => webpack(platform, options.env),
         ];
     }
@@ -68,14 +70,14 @@ function execute(options) {
     return commands.reduce((current, next) => current.then(next), Promise.resolve());
 }
 
-function cleanBuildArtifacts(platform) {
+function cleanBuildArtefacts(platform) {
     return new Promise((resolve, reject) => {
         if (platform !== "android") {
             return resolve();
         }
 
         getTnsVersion().then(version => {
-            // the android build artifacts should be cleaned manually
+            // the android build artefacts should be cleaned manually
             // for nativescript-cli v3.0.1 and below or if using uglify
             if (isVersionGte(version, "3.0.1") || shouldUglify()) {
                 gradlewClean().then(resolve).catch(throwError);
@@ -96,7 +98,7 @@ function installSnapshotArtefacts() {
 
 function gradlewClean() {
     return new Promise((resolve, reject) => {
-        const platformsPath = pathResolve(PROJECT_DIR, "platforms", "android")
+        const platformsPath = join(PROJECT_DIR, "platforms", "android")
         const gradlew = pathResolve(platformsPath, "gradlew");
         if (!existsSync(gradlew)) {
             return resolve();
@@ -134,6 +136,7 @@ function versionToNumber(version) {
 // Clear platform/**/app folder contents
 function cleanApp(platform) {
     return new Promise((resolve, reject) => {
+
         spawnChildProcess("tns", "clean-app", platform)
             .then(resolve)
             .catch(throwError)
@@ -171,18 +174,14 @@ function runTns(command, platform) {
 }
 
 function getOptions(flags) {
-    let options = {};
-    options.platform = getPlatform(flags);
-    options.command = getCommand(flags);
-    options.env = getEnv(flags);
-    options.bundle = !flags.includes("nobundle");
-
-    return options;
+    return {
+        platform: getPlatform(flags),
+        command: getCommand(flags),
+        env: flags.filter(isEnvCommand),
+        bundle: !flags.includes("nobundle"),
+    };
 }
 
-function getEnv(flags) {
-    return flags.filter(item => isEnvCommand(item));    
-}
 
 function getPlatform(flags) {
     if (flags.includes("android") && flags.includes("ios")) {
@@ -194,7 +193,9 @@ function getPlatform(flags) {
     } else if (flags.includes("ios")) {
         return "ios";
     } else {
-        throwError({message: "You must provide a target platform! Use either --android, or --ios flag."});
+        throwError({message:
+            "You must provide a target platform! " +
+            "Use either --android, or --ios flag."});
     }
 }
 
@@ -209,7 +210,10 @@ function getCommand(flags) {
 
 function spawnChildProcess(command, ...args) {
     return new Promise((resolve, reject) => {
-        const childProcess = spawn(command, args, {
+        const escapedArgs = args.map(escapeWithQuotes)
+        const escapedCommand = escapeWithQuotes(command)
+
+        const childProcess = spawn(escapedCommand, escapedArgs, {
             stdio: "inherit",
             pwd: PROJECT_DIR,
             shell: true,

--- a/bin/ns-bundle
+++ b/bin/ns-bundle
@@ -79,7 +79,7 @@ function cleanBuildArtefacts(platform) {
         getTnsVersion().then(version => {
             // the android build artefacts should be cleaned manually
             // for nativescript-cli v3.0.1 and below or if using uglify
-            if (isVersionGte(version, "3.0.1") || shouldUglify()) {
+            if (!isVersionGte(version, "3.0.1") || shouldUglify()) {
                 gradlewClean().then(resolve).catch(throwError);
             } else {
                 return resolve();
@@ -155,7 +155,7 @@ function webpack(platform, env) {
             ...env.map(item => `--${item}`),
             shouldUglify() && `--env.uglify`,
             shouldSnapshot(platform) && `--env.snapshot`
-        ];
+        ].filter(a => !!a);
 
         spawnChildProcess(...args)
             .then(resolve)

